### PR TITLE
Throw exception when exceeding maximum number of Sql Server parameters

### PR DESCRIPTION
**Pull request by othercorey**
Refs https://github.com/cakephp/cakephp/issues/14775

### CODE REVIEW

File: src/Database/Driver/Sqlserver.php [hunk 1]
```diff
@@ -192,17 +193,22 @@ public function enabled(): bool
...
         $this->connect();
+
         $options = [PDO::ATTR_CURSOR => PDO::CURSOR_SCROLL];
-        $isObject = $query instanceof Query;
-        /** @psalm-suppress PossiblyInvalidMethodCall */
-        if ($isObject && $query->isBufferedResultsEnabled() === false) {
-            $options = [];
+        if ($query instanceof Query) {
+            if (count($query->getValueBinder()->bindings()) > 2100) {
+                throw new Exception(
```
Comment (ADmad) [inline src/Database/Driver/Sqlserver.php:200]: Using core exception feels a bit weird here but we don't have a generic database exception :(

Comment (othercorey) [reply src/Database/Driver/Sqlserver.php:200]: I don't know when to use PHP exceptions and when to use Cake exceptions.

Comment (markstory) [reply src/Database/Driver/Sqlserver.php:202]: We often use basic php exceptions when the class fits the situation and we don't have an need to provide a special error page.

**Comment by othercorey** [comment]
Changed exception to InvalidArgumentException and updated explanation of the possible causes.

**Pull request merged**